lib/rust/ensogl/core/src/display/style/data.rs

```diff
@@ -47,16 +50,25 @@ where color::Color<C>: Into<color::Rgba>
     }
 }
 
-impl TryFrom<String> for Data {
-    type Error = ();
-    fn try_from(s: String) -> Result<Self, Self::Error> {
-        match s.parse::<f32>() {
-            Ok(t) => Ok(Data::Number(t)),
-            _ => match s.parse::<color::AnyFormat>() {
-                Ok(t) => Ok(Data::Color(t.into())),
-                _ => Err(()),
-            },
+impl From<&str> for Data {
```

Comment:
Why we tried from String, but now we silently copy &str?

Reply:
As we discussed, I added a comment explaining that the conversion is needed to allow entering text literals as values in the hardcoded theme. The parsing that was previously done in the `TryFrom<String>` implementation is now migrated to the `FromStr` implementation.